code/src/components/ProjectToDoList.js

```diff
@@ -0,0 +1,221 @@
+
+export const ProjectToDoList = () => {
+  const [projectCreated, setProjectCreated] = useState(false);
+  const [listNameCreated, setListNameCreated] = useState(false);
+  const [listName, setListName] = useState('');
+  const [itemName, setItemName] = useState('');
+  const [dueDate, setDueDate] = useState('');
+  const [showForm, setShowForm] = useState(false);
+  const [projectActive, setProjectActive] = useState(true);
+
+  const lists = useSelector((state) => state.projects.lists);
+
+  const dispatch = useDispatch();
+
+  const handleAddList = () => {
+    dispatch(addList({ name: listName.charAt(0).toUpperCase() + listName.slice(1) }));
+    setListName('');
+    setListNameCreated(true)
+    setProjectCreated(false)
+  };
+
+  const handleAddItem = (listIndex) => {
+    const newToDo = {
+      id: uniqid(),
+      text: itemName[listIndex].charAt(0).toUpperCase() + itemName[listIndex].slice(1),
+      due: dueDate ? dueDate.toDateString() : '-',
+      complete: false,
+      priority: false
+    };
+    dispatch(addItem({ listIndex, item: newToDo }));
+    setItemName('');
+  };
+
+  const handleWantNewProject = () => {
+    !projectCreated ? setProjectCreated(true) : setProjectCreated(!projectCreated)
+  };
+
+  const handleItemNameChange = (listIndex, value) => {
+    setItemName({ ...itemName, [listIndex]: value });
+  }
+
+  const handleProjectClick = (listIndex) => {
+    setProjectActive((prevProjectActive) => ({
+      ...prevProjectActive,
+      [listIndex]: !prevProjectActive[listIndex],
+    }));
+  };
+
+  const onToDoToggle = (id, listIndex) => {
+    dispatch(toggleItem({ listIndex, itemId: id }));
+    dispatch(sortItems({ listIndex }));
+    dispatch(sortPriority({ listIndex }));
+}
+
+  const onTogglePriority = (id, listIndex) => {
+    dispatch(togglePriority({ listIndex, itemId: id }));
+    dispatch(sortPriority({ listIndex }));
+    dispatch(sortItems({ listIndex }));
+  };
+
+  const onToDoDelete = (id, listIndex) => {
+    dispatch(deleteItem({ listIndex, itemId: id }));
+  };
+
+  const deleteListByName = (list) => {
+    dispatch(deleteList({ name: list.name }));
+  }
+
+  const handleShowForm = () => {
+    setShowForm(true)
+  }
+
+  const handleHideForm = () => {
+    setShowForm(false)
+  }
+
+  const colorsArray = ['#3d5a80', '#b8dee0', '#79c9b0', '#e8e863', '#e4d1f1', '#831e74', '#ffccb8']
+  const a = Math.floor(Math.random() * colorsArray.length);
+  
+  return (
+    <>
+      {listNameCreated && (
+        <ProjectWrapper>
+          {Array.isArray(lists) && lists.map((list, listIndex) => (
+            <>
+              <ProjectHeader key={list.name} style={{ background:`${colorsArray[a, listIndex]}`}}>
```

Comment:
A tip is to use key prop on the outermost component: In the Array.map() method, the key prop is currently being set on the ProjectHeader component. It is recommended to set it on the outermost div or fragment component instead. :)